spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/channel/JsonMessageRowMapper.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.store.channel;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.springframework.integration.support.json.JacksonJsonObjectMapper;
+import org.springframework.integration.support.json.JacksonMessagingUtils;
+import org.springframework.integration.support.json.JsonObjectMapper;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link RowMapper} implementation that deserializes {@link Message} objects from
+ * JSON format stored in the database.
+ * <p>
+ * This mapper works in conjunction with {@link JsonChannelMessageStorePreparedStatementSetter}
+ * to provide JSON serialization for Spring Integration's JDBC Channel Message Store.
+ * <p>
+ * Unlike the default {@link MessageRowMapper} which uses Java serialization,
+ * this implementation uses JSON to deserialize message strings from the MESSAGE_CONTENT column.
+ * <p>
+ * The underlying {@link JsonObjectMapper} validates all deserialized classes against a
+ * trusted package list to prevent security vulnerabilities.
+ * <p>
+ * <b>Usage Example:</b>
+ * <pre>{@code
+ * &#64;Bean
+ * JdbcChannelMessageStore messageStore(DataSource dataSource) {
+ *     JdbcChannelMessageStore store = new JdbcChannelMessageStore(dataSource);
+ *     store.setChannelMessageStoreQueryProvider(new PostgresChannelMessageStoreQueryProvider());
+ *
+ *     // Enable JSON serialization
+ *     store.setPreparedStatementSetter(
+ *         new JsonChannelMessageStorePreparedStatementSetter());
+ *     store.setMessageRowMapper(
+ *         new JsonMessageRowMapper("com.example"));
+ *
+ *     return store;
+ * }
+ * }</pre>
+ *
+ * @author Yoobin Yoon
+ *
+ * @since 7.0
+ */
+public class JsonMessageRowMapper implements RowMapper<Message<?>> {
+
+	private final JsonObjectMapper<?, ?> jsonObjectMapper;
+
+	/**
+	 * Create a new {@link JsonMessageRowMapper} with the default {@link JsonObjectMapper}
+	 * configured for Spring Integration message deserialization.
+	 * <p>
+	 * This constructor is suitable when deserializing standard Spring Integration
+	 * and Java classes. Custom payload types will require their package to be
+	 * specified using {@link #JsonMessageRowMapper(String...)}.
+	 */
+	public JsonMessageRowMapper() {
```

Comment:
I think we don't need this ctor.
And another one could be like `String @Nullable ... trustedPackages`